src/bootstrap/lib.rs

```diff
@@ -628,8 +628,10 @@ impl Build {
             self.run(&mut update(false));
         }
 
+        self.run(Command::new("git").args(&["stash", "save"]).current_dir(&absolute_path));
         self.run(Command::new("git").args(&["reset", "-q", "--hard"]).current_dir(&absolute_path));
-        self.run(Command::new("git").args(&["clean", "-qdfx"]).current_dir(absolute_path));
+        self.run(Command::new("git").args(&["clean", "-qdfx"]).current_dir(&absolute_path));
+        self.run(Command::new("git").args(&["stash", "apply"]).current_dir(absolute_path));
```

Comment:
I guess `git stash pop` is more useful here:
> apply [--index] [-q|--quiet] [<stash>]
Like pop, but do not remove the state from the stash list. Unlike pop, <stash> may be any commit that looks like a commit created by stash push or stash create.

I don't want to have a new stash item whenever running `x.py`.

Please also keep in mind that applying a stash can fail due to merge conflicts!